### ISSUE TITLE
Dropdown List doesn't update when jquery-sb is applied

### ISSUE THREAD

**Issue by BeachcomberCo** [revsystems/jQuery-SelectBox#56]
In an asp page I have there is an sb selectbox:
<select name="model" id="model" size="1" class="enqLargeSelect" onChange="document.getElementById('UpdateSave').style.display='inline';">
with a loop that loads up a bunch of <option entries.
This results, in the HTML, in an unordered list of these entries
<ul class="selectbox items enqLargeSelect" id="sbdd670770964" role="menu" aria-hidden="true" style="left: 214.9px; top: 415px; display: none; visibility: visible; position: absolute; max-height: 275px;" TAX9="391">

with a bunch of <li elements:
<li class="selected" id="sbo176171601" role="option" aria-disabled="" TAX9="398">

Following that is a select box:
<select name="model" class="enqLargeSelect has_sb" id="model" style="display: inline-block;" onchange="document.getElementById('UpdateSave').style.display='inline';" size="1" TAX9="482">

with its <option elements.

So far so good.  
When you change the selected value, a call is made resulting in a pop-up window asking for some further options/information to be selected/given.

One of the things that should happen when you OK that window is that the select box selectedindex be set to 0.  To that end, there is a function in that pop-up that does:

```
    parent.window.document.custnewform.model.selectedIndex = 0;
```

However, this achieves nothing, apparently.  

In the header of the parent asp I have:
            $("select").sb({
                "fixedWidth": true,
                "maxHeight": '300px',
                useTie:true
            });
and in the header is:
    <script type="text/javascript" src="jquery.tie.js?v=<%=Application("version")%>"></script>

The <li class="selected" is unchanged (still what it was before opening the pop-up window) and the same option is still set as selected in the dropdown box.

I'm not familiar with the sb or, for that matter, particularly familiar with javascript/html etc., I'm afraid.

Could anyone tell me what I need to do to get this to work, please?
Thanks
Steve

**Comment by BeachcomberCo** [comment]
Hmm - that didn't come out too well, did it?  With a bunch of refers to li elements.
and in the header is:  refers to an include file for the jquery.tie.js

**Comment by MaxPower15** [comment]
Hey there. The "tie" plugin only latches on to jQuery actions/events, and it looks like you're using non-jQuery methods to set the value. You might try manually triggering a "change" event on the select after setting it.

**Comment by BeachcomberCo** [comment]
OK, thanks - that's useful.  How might I trigger a change event, please?
Thanks
Steve

**Comment by MaxPower15** [comment]
It kind of depends on the framework you're using, but it usually looks like $("#my_select").trigger("change"). If you're not familiar with jQuery, that syntax might be a little confusing--I recommend reading up! :)

**Comment by BeachcomberCo** [comment]
Thanks very much for that.  I think I have some reading to do.  The code I have at the moment is something like:
    function savesettings() {
        parent.window.document.custnewform.newcontact.value = document.aForm.t1.value;
        parent.window.document.custnewform.source.value=document.aForm.I_source.value;
        parent.window.document.custnewform.newsource.value=document.aForm.I_source.value;
        parent.window.document.custnewform.model.selectedIndex = 0;

I'm guessed that I need something like:
parent.window.document.custnewform.model.trigger("change");
but that doesn't work (no trigger).
This is all in classic asp, javascript and vbscript :-(

**Comment by BeachcomberCo** [comment]
I tried $("#model").trigger("change"); which didn't raise an error, but also didn't do anything.

**Comment by MaxPower15** [comment]
Yeah, it's hard to say what's the right thing to do without seeing the markup. If I had to guess, assuming this isn't use any special framework, it might be `$("form[name=custnewform] select[name=model]").trigger("change")`.

**Comment by BeachcomberCo** [comment]
No special framework, AFAIK.  Thanks for the suggestion - tried it but no difference, I'm afraid.  the li element in the ul is still given a class="selected", and the option in the dropdown list is still set as selected.  And the text of the sb subdivision still has the original value.  
So following on from your suggestion I have tried a couple of other things.  
I have put a call to another function in the parent window:
window.parent.triggerModel();

and the function is:

```
function triggerModel(){
    document.getElementById('model').selectedIndex = 0;
    $(document).trigger('change');
}
```

But this, too, fails to do anything useful.  I tried changing that function to include the line:
$("form[name=custnewform] select[name=model]").trigger("change")
but still nothing.
